app/models/bookmark.rb

```diff
@@ -1,3 +1,10 @@
+class ControllerValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    controllers = ["dashboard"] + (ActiveRecord::Base.connection.tables.map(&:to_s) + Permission.resources.map {|x| x.tableize }).uniq
```

Comment:
Do you why have I removed https://github.com/theforeman/foreman/pull/2811/files#diff-a18735a12ab93f58a5ad67cf81a783ecR14 ? I can revert that and invalidate on permission.create. I that what you mean? or do you want it all to be added to a `Rails.cache`?